### DISCUSSION TITLE
Diagonal roads

### DISCUSSION THREAD

**Discussion by victorleblais** [voxeltycoon/issues#338] [General]
**Is your feature request related to a problem? Please describe.**
One of my biggest gripe with OpenTTD and other building games is the lack of diagonal roads. The same "issue" is present here in Voxel Tycoon.

**Describe the solution you'd like**
Diagonal roads. One could imagine a building system similar to how rails work.
Quite unsure how intersection would be handled since they could become quite complex.
The worst-case scenario would be "merging" a 4-way crossing using straight roads with a 4 way crossing using diagonal roads, making it a 8-way "star" intersection. Maybe a solution to make it simpler (mainly about the path that vehicles use) would be to automatically create a small round-about when more than *x* roads cross each other...

**Describe alternatives you've considered**
I'm considering sacrificing a bucket of asphalt to the road gods for this to happen.

**Additional context**
I totally understand that making diagonals on a tile-based game is pretty tricky, especially for roads, where you have additional constraints (keep proportions / size of lanes the same, buildings to the side of roads, complex crossings, etc...). I just hope that the dev team can figure out a way to do it. It would make a big difference compared to other titles in the genre.

I'm making this suggestion since it's something that I think will be a popular request.

**Comment by StolleJay** [comment]
Please yes.
There are diagonal rails so why not also diagonal roads?

**Comment by victorleblais** [comment]
Announced as planned in the roadmap and FAQ:
![image](https://user-images.githubusercontent.com/4365075/115476826-ef398e80-a242-11eb-8ec1-f33986772971.png)
https://voxeltycoon.xyz/devlog/ea-faq/
https://voxeltycoon.xyz/roadmap
